### ISSUE TITLE
Question about the ephemeris in toa.TOA

### ISSUE THREAD

**Issue by matteobachetti** [nanograv/PINT#1040]
Hi, 
I don't understand how to specify the ephemeris used to obtain a given barycentered TOA when creating a `TOA` object in the 'tdb' timescale. If the system is tdb, do I have to worry about the ephemeris at all? Or, once the tdb was calculated, this is independent?
I know this should be something obvious, but I'm never sure that I'm not missing important details.
Thanks in advance

**Comment by paulray** [comment]
If you create a barycentered TOA in the TDB timescale, then all of the conversions have already been done, including the timescale change from TT to TDB and any solar system delays.  There isn't really any way of recording what solar system ephemeris or time ephemeris was used for that conversion.  For example, if you used barycorr to do that, you could have used any JPL ephemeris version, and it uses FB90 for the time ephemeris.  In any event, PINT will just treat those as barycentered TDB times. An issue might come up if you try to fit topocentric TOAs along with your barycentered ones.  For full accuracy, you should have PINT used the same JPL ephem and time ephem as was used for the barycentric TOAs, but there is not automatic way to ensure this.  Does that help?

**Comment by matteobachetti** [comment]
I'm dealing with barycentric TOAs produced through different procedures and, probably, ephemeris, by different people. I guess this means that I have to expect large (~ms) differences between them, right?

**Comment by paulray** [comment]
Shouldn't be ms. Depending on how old the ephemerides are, I wouldn't expect more than µs differences.

**Comment by matteobachetti** [comment]
Thanks a lot. Do you have any docs about the maximum deviations to be expected between, say, DE200 vs DE430?

**Comment by scottransom** [comment]
Hmmm.  For TDB conversion I would expect a much smaller difference than $\mu$s.  Maybe even at the ns level given what is changing between them (it is just a correction for the overall grav redshift of the Solar system, right?  And so differences between the ephemerides must change that redshift by a *tiny* amount.)

**Comment by scottransom** [comment]
Doesn't it almost have to be super tiny by definition since it is the time unit used to *access* the JPL ephemerides?  https://ssd.jpl.nasa.gov/?faq#B06

**Comment by paulray** [comment]
See also this (now quite old) comparison of ephemerides in Fig 6 of this paper: https://ui.adsabs.harvard.edu/abs/1994ApJ...428..713K/abstract

**Comment by scottransom** [comment]
@matteobachetti You are concerned with the differences in *TDB times* between ephemerides, right?  Not in differences between timing solutions?  If so, I don't think that old Kaspi et al paper gives us much information.

**Comment by aarchiba** [comment]
Be aware that barycentered TOAs, for example the usual data from satellites, will have been corrected for light propagation time across the solar system, which depends much more sensitively on the ephemeris used (and the source position on the sky).

**Comment by matteobachetti** [comment]
Ok, I just made a test using many NuSTAR observations of the same object over time, correcting the event arrival times with barycorr using various ephemerides. DE421 and DE430 are not that different, but DE200 definitely is. The problem is that different packages use different ephemerides by default, and I think some of the people who provided the data got confused or forgot some command line switch.
For the correction, I always used the same coordinates, but it looks like the difference between the ICRS and FK5 coordinates is tiny anyway (well below the arcsecond)
![eph_difference](https://user-images.githubusercontent.com/7190189/117432446-acf29b80-af2a-11eb-9913-b85c75b372ce.png)

**Comment by paulray** [comment]
Yeah, no one should use DE200 for anything anymore.  It is really bad that FTOOLs like `barycorr` still have DE200 as the default if you don't say anything else.
